std/utf.d

```diff
@@ -1627,12 +1752,47 @@ version(unittest) private void testDecodeFront(R)(ref R range,
     }
 }
 
-version(unittest) private void testBothDecode(R)(R range,
+version(unittest) private void testDecodeBack(R)(ref R range,
                                                  dchar expectedChar,
-                                                 size_t expectedIndex,
+                                                 size_t expectedNumCodeUnits,
                                                  size_t line = __LINE__)
 {
+    static if (!isSomeString!R || !isBidirectionalRange!R || (isRandomAccessRange!R && !hasLength!R))
+        return;
```

Comment:
Ok, it isn't nice at all, but it is here, because I wanted to avoid creating the whole new unit tests just for `decodeBack` which would look exactly like test for `decode` and `decodeFront`, but omitting `InputRange`. If there is a better way or I should separate them anyway, I would kindly do so.